### ISSUE TITLE
streaming over internet

### ISSUE THREAD

**Issue by boostjunkie92** [moonlight-stream/moonlight-docs#52]
Having an issue with my phone not seeing my computer over data connection. within the home network works fine but when i switch to mobile data the computer cannot be detected.  Ive forwarded the necessary ports from my PC with a static ip adress within my LAN as well as set the external IP to "any". when i go to look up my external IP at the provided website i get a longer IP address with semicolons and letters. not sure if my setup is correct.

**Comment by cgutman** [comment]
That IP address you got is an IPv6 address. You'll need to find your IPv4 address to work with Moonlight. Try a few other sites that tell you your IP address and you should find one that looks in the form: 123.123.123.123

**Comment by boostjunkie92** [comment]
when i went to whatismyip.org they gave me a 68.xxx ip address which i figured was the external IP but still didnt work

**Comment by reinaldorauch** [comment]
Why ipv6 doesn't work? the shield protocol or something like that doesn't allow it?

**Comment by cgutman** [comment]
GeForce Experience listens on IPv4 only.

**Comment by reinaldorauch** [comment]
@cgutman but if a rule on windows for example to forward the ports from the ipv6 address to the ipv4 it would work? the platform would allow it?

**Comment by Lukestifire** [comment]
Was there ever a resolution to this?
I have tried everything to get this to work. I've fresh installed everything twice. I forwarded all listed ports and created a static IP. It works fine when I'm on the same network but as soon as I try to stream to another PC or phone over the internet it just won't connect. There is no more info on the internet, I've been searching for days. 

Anyone have any idea what else I can do?

**Comment by cgutman** [comment]
@Lukestifire you should make sure your forwarding rules are correct and that you aren't double-NATed (connected behind 2 or more routers).

**Comment by Lukestifire** [comment]
@cgutman I'm only connected through xfinity's combo modem router. The port forwarding options are on the myXfi website now. I have the port numbers specified to TCP and UDP respectively. 

Port forwarding: https://i.imgur.com/GanqQCB.png

Thhats a screenshot of it. It's strange that it makes me choose a device. The device listed at the top 
"TP-LinkT-6B98" is my Gaming PC that's connected to the router with an Ethernet cable.

Thanks for your help btw

**Comment by cgutman** [comment]
What happens when you try to connect? Does the PC show up as online and then fail trying to stream or does it appear offline?

What happens when you press Add PC and specify the address from https://whatsmyip.com/ (run on your home PC)?

**Comment by Lukestifire** [comment]
When I try to connect I see this : https://i.imgur.com/gKLUtuJ.png

If I add my IP (I tried both IP and LAN) It says it fails. So it sees the pc it just won't let me access it...

**Comment by cgutman** [comment]
On your home network, can you go to https://www.yougetsignal.com/tools/open-ports/ and check port 47989 and 47984? They should both be open if the forwarding is working correctly.

**Comment by Lukestifire** [comment]
OK. So... It works now. THANKYOU for that port checking thing. The port wasn't open.

 I went and verified that that TP link thing was actually my desktop and it turned out the IP did not match. So I found the correct device in the list with a matching IP and changed the port forwarding list to that device and it worked!

But it's so laggy and slow I can't seem to do anything. I have 30 meg download and 5 meg up connection on my host and client internet connection. Even with the settings in moonlight set to 5 megs its pretty bad.
